### DISCUSSION TITLE
Designating some part of template file content to not have "template" functionality??

### DISCUSSION THREAD

**Discussion by jfines** [SilentVoid13/Templater#1198] [Help]
This quest originated because of a desire to have frontmatter that is unique to or applies only to the template file/s one might have (template functionality not being applied to it) as well as frontmatter in that template file that get's applied to the note made from it (template functionality being applied). When I first [tried to grapple with the idea](https://forum.obsidian.md/t/can-you-have-frontmatter-in-a-template-that-pertains-to-the-template-itself-and-not-what-comes-from-the-template-can-you-have-both-how/67528) I struggled a bit to articulate what I was asking for; but, eventually, I was able to [firm up my question](https://forum.obsidian.md/t/frontmatter-in-templater-seeking-solutions-for-3-use-cases/67670).

Isn't there SOME way to exclude template functionality from some, designated part of the template's content? It would allow to have frontmatter for the templates themselves (with their own key values for the template file/s) AND for the notes made from the templates (with their own keys and values). It would be nice to have frontmatter for your templates (ie: the template files themselves).

If anyone knows of or can suggest any way to do this (whether with Templater or by using the Obsidian API or both) please do let me know.

Yes **_I would love to have any existing solution_**, but, in addition to that...

If functionality WERE to be implemented in Templater for this - here is an example I thought of that might work.

Encapsulate the entire body of Templater content within an Obsidian file. One way this could be done is similar to how other plugins have done. Use back ticks / code block together with a key word "templater" to designate the section of the file that should be treated as a template. Here is an example of what that might look like.

```
    // Non Templater code goes here

    ```templater

        // Templater code goes here

    ```

```

Interestingly enough this would allow for more complex scenarios fitting the paradigm. Such as..

```
    // Non Templater code here

    ```templater

        // Templater code here

    ```

    // More non Templater code

```

Or even...

```
    // Non Templater code here

    ```templater

        // Templater code here

    ```

    // More non Templater code

    ```templater

        // More Templater code

    ```

```

Specifically you could definitely do this...

```
    ---
    title: "some template"
    category: "the template category"
    tags: [templates]
    link: "[[some other template]]"
    comment: this frontmatter is for the file and doesn't show up in the note made from the template
    ---

    ```templater

        ---
        title: "my note title"
        type: "my note type"
        date: XX-XX-XXXX
        comment: this frontmatter is for the template and does show up in the note
        ---
        
        tp.date.now(MM-DD-YYYY)
        // Additional Templater code...
    ```

```

I probably got the syntax wrong since I'm still learning but I think it conveys the idea (I hope).

Any help much appreciated.

**Comment by Zachatoo** [comment]
How would this proposed solution impact already existing templates? Would all templates be required to update to be enclosed in triple backticks in order to be executed? Would there be any way to allow for inline templating still? In my mind, this feels like it would cause a lot of pain for a lot of Templater users to have to switch to this new syntax.

To answer your question, assuming that you only need separate frontmatter for your templates, you can clear out anything before a certain point using `<%* tR = "" -%>`.

```js
---
title: "some template"
category: "the template category"
tags:
  - templates
link: "[[some other template]]"
comment: this frontmatter is for the file and doesn't show up in the note made from the template
---
<%* tR = "" -%>
---
title: "my note title"
type:
  - my note type
date: <% tp.date.now() %>
comment: this frontmatter is for the template and does show up in the note
---
// rest of template
```

**Reply by Zachatoo** [reply]
[Link](https://silentvoid13.github.io/Templater/commands/execution-command.html#how-to-output-a-value-from-a-javascript-execution-command-) to relevant docs. It's not super well documented how `tR` works, but that is where it is mentioned and how I found this solution.

**Reply by jfines** [reply]
@Zachatoo 
> How would this proposed solution impact already existing templates? Would all templates be required to update to be enclosed in triple backticks in order to be executed? Would there be any way to allow for inline templating still?

What's wrong with having a config item for it?

In fact - how config items function is limited only by the imagination.

**Reply by Zachatoo** [reply]
There is such thing as too much configuration.

- More development maintenance.
   - Having to maintain multiple versions that do similar things.
- More cognitive load for end users.
   - What does this config do again?
   - Do I need this?
- More opportunities for user error.
   - Why isn't this working? Why can no one else reproduce this issue? Oh, it's because I turned this toggle without noticing and no one else did. That took 3 hours to figure out instead of 5 minutes.

I wouldn't push for more configuration unless it had a really strong case. I don't consider this a strong case because it's not often requested, and there's a workaround.

Does the workaround I provided work for you?

**Reply by jfines** [reply]
Sounds like you've got your mind made up.
